lightning/config/config.go

```diff
@@ -204,8 +205,68 @@ func (cfg *Config) LoadFromGlobal(global *GlobalConfig) error {
 }
 
 // LoadFromTOML overwrites the current configuration by the TOML data
+// If data contains toml items not in Config and GlobalConfig, return an error
+// If data contains toml items not in Config, thus won't take effect, warn user
 func (cfg *Config) LoadFromTOML(data []byte) error {
-	return errors.Trace(toml.Unmarshal(data, cfg))
+	// warnItems saves legal toml items but won't effect
+	var warnItems []string
+	dataStr := string(data)
+
+	// Here we load toml into cfg, and rest logic is check unused keys
+	metaData, err := toml.Decode(dataStr, cfg)
+
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	unusedConfigKeys := metaData.Undecoded()
+	if len(unusedConfigKeys) == 0 {
+		return nil
+	}
+
+	// Now we deal with potential both-unused keys of Config and GlobalConfig struct
+
+	var bothUnused []string
+	// Key type returned by metadata.Undecoded doesn't have a equality comparison,
+	// we convert them to string type instead, and this conversion is identical
+	unusedConfigKeyStrs := make([]string, len(unusedConfigKeys))
+	for i, key := range unusedConfigKeys {
+		unusedConfigKeyStrs[i] = key.String()
+	}
+
+	globalCfgDummy := NewGlobalConfig()
+	metaDataGlobal, errTmp := toml.Decode(dataStr, &globalCfgDummy)
+	if errTmp != nil {
+		return errors.Trace(errTmp)
+	}
+
+	unusedGlobalKeys := metaDataGlobal.Undecoded()
+	unusedGlobalKeyStrs := make(map[string]struct{})
+	for _, key := range unusedGlobalKeys {
+		unusedGlobalKeyStrs[key.String()] = struct{}{}
+	}
+
+	for _, keyStr := range unusedConfigKeyStrs {
+		if _, found := unusedGlobalKeyStrs[keyStr]; found {
+			bothUnused = append(bothUnused, keyStr)
+		} else {
+			warnItems = append(warnItems, keyStr)
+		}
+	}
+
+	if len(bothUnused) > 0 {
+		err = errors.New(fmt.Sprintf("config file contained unknown configuration options: %s",
+			strings.Join(bothUnused, ", ")))
+		return errors.Trace(err)
```

Comment:
No need to `errors.Trace` any errors created using `errors.New` or `errors.Errorf`.

```suggestion
		return err
```